custom/icds_reports/utils/aggregation_helpers/distributed/agg_awc.py

```diff
@@ -283,9 +283,13 @@ def updates(self):
                     ucr.supervisor_id = adolescent_girls_table.supervisor_id AND
                     adolescent_girls_table.month=%(start_date)s
                     )
-            WHERE (opened_on <= %(end_date)s AND
-              (closed_on IS NULL OR closed_on >= %(start_date)s )) AND
-              migration_status IS DISTINCT FROM 1
+            WHERE (
```

Comment:
shouldnt the inner join on the AG table handle this? This is already limited to open AG cases

Reply:
yeah, inner join can handle this and would be equivalent to this where clause.
Just for my curiosity. Is there a performance gain we get by putting it in inner join?

Reply:
I meant some of this may be redundant. We already only include open AG cases, are we trying to exclude specific open AG cases? It seems like a lot of the filtering you have added is already done by default when we limit to the cases that exist in the ag_registration table.

Reply:
hey that adolescent_girls table is a table of ag forms, not the cases themselves. Yes, those forms are filled for cases only.  But this filter it to avoid a case where a ag form is filled 4 months back for a girl who was in AG age limit then but is not now. So now she should be excluded because she is more than 14 years. This check is to apply that